recipes/system-probe.rb

```diff
@@ -20,12 +20,15 @@
 is_windows = platform_family?('windows')
 
 # Set the correct agent startup action
+sysprobe_enabled = node['datadog']['system_probe']['enabled']
+npm_enabled = node['datadog']['system_probe']['network_enabled'] 
+usm_enabled = node['datadog']['system_probe']['service_monitoring_enabled']
 cws_enabled = node['datadog']['security_agent']['cws']['enabled']
-sysprobe_enabled = if is_windows
-                     node['datadog']['system_probe']['network_enabled']
-                   else
-                     node['datadog']['system_probe']['enabled'] || node['datadog']['system_probe']['network_enabled'] || node['datadog']['system_probe']['service_monitoring_enabled'] || cws_enabled
-                   end
+
+sysprobe_enabled ||= npm_enabled || cws_enabled
+unless is_windows
```

Comment:
Windows has USM; so on windows system probe should be enabled if usm is enabled.

(this is also somewhat outside the scope of what you're doing, because it's clear the USM support for windows is lacking in the chef recipe).  So, I'm fine either way.  But there's no underlying reason for the conditional at all.

Reply:
Ah didn't know that, yeah the USM is checked here but not plugged anywhere after that. I will simplify further then